iocage_cli/activate.py

```diff
@@ -21,22 +22,49 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
 # IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 # POSSIBILITY OF SUCH DAMAGE.
-"""activate module for the cli."""
+"""Activate zfs pools for iocage with the CLI."""
 import click
 
-import iocage_lib.ioc_common as ioc_common
-import iocage_lib.iocage as ioc
+import iocage.errors
+import iocage.Datasets
+import iocage.Logger
+import iocage.ZFS
 
 __rootcmd__ = True
 
 
 @click.command(name="activate", help="Set a zpool active for iocage usage.")
+@click.pass_context
 @click.argument("zpool")
-def cli(zpool):
-    """Calls ZFS set to change the property org.freebsd.ioc:active to yes."""
-    ioc.IOCage(activate=True).activate(zpool)
-
-    ioc_common.logit({
-        "level"  : "INFO",
-        "message": f"ZFS pool '{zpool}' successfully activated."
-    })
+@click.option("--mountpoint", "-m", default="/iocage")
+def cli(ctx, zpool, mountpoint):
+    """Call ZFS set to change the property org.freebsd.ioc:active to yes."""
+    logger = ctx.parent.logger
+    zfs = ctx.parent.zfs
+
+    if ctx.parent.user_sources is not None:
+        logger.error("Cannot activate when executed with explicit sources.")
+        exit(1)
+
+    iocage_pool = None
+
+    for pool in zfs.pools:
+        if pool.name == zpool:
+            iocage_pool = pool
+
+    if iocage_pool is None:
+        logger.error(f"ZFS pool '{zpool}' not found")
+        exit(1)
+
+    try:
+        datasets = iocage.Datasets.Datasets(
+            zfs=zfs,
+            logger=logger
+        )
+        datasets.activate_pool(
+            pool=iocage_pool,
+            mountpoint=mountpoint
+        )
+        logger.log(f"ZFS pool '{zpool}' activated")
+    except iocage.errors.IocageException:
```

Comment:
Shouldnt we print something to the user in case that happens? Does that exception have an error message?

Reply:
The default logger always prints `error` and `warn` levels, so that each such error will already have printed to the users screen using the CLI. 

When an `iocage.error.IocageException` occurs, we know that the user was notified already.